lib/jira/client/index.js

```diff
@@ -141,6 +142,8 @@ module.exports = async (jiraHost, gitHubInstallationId, logger) => {
   return client
 }
 
+module.exports = async (...args) => newrelic.startSegment('lib/jira/client: getJiraClient', true, async () => getJiraClient(...args))
```

Comment:
@tebriel and I talked on Zoom about how this is more appropriate to have the tracing code be part of the function definition rather than the consuming code. I totally agree with him.

This isn't the prettiest code, but it's a reusable away for us to wrap tracing external to the function definition. Rather than pushing it INSIDE the definition of `getJiraClient` here. I could see this potentially being abstracted to be reused across the codebase.

Reply:
Great work! I think once we get a few more of these under our belt we can probably create a function to wrap this to make it prettier. like `wrapAsync(name, func)` thinking like the following:

```js
module.exports = NRWrapAsync('lib/jira/client: getJiraClient', getJiraClient);
```

But not important now and doesn't really apply yet!

Reply:
Yeah 100%. I'm sure we'll start reusing this very soon, and I'm all for a way to push this cruft into one spot instead of copy/paste boilerplate all over.